my-app/src/components/Doctors/AddData.js

```diff
@@ -1,61 +1,72 @@
-import React, { Component } from "react";
+import React, { useState, useEffect } from 'react';
+import firebase from '../Firebase/firebase'
+import AddImage from '../Firebase/ImageUpload'
+import { Button, Header, Modal, Image } from 'semantic-ui-react'
+import { useTranslation } from 'react-i18next';
+import i18next from 'i18next';
 
 
-const DEFAULT_STATE = {
-  firstname: "",
-  surname: "",
-  vetClinic: ""
-};
 
-export default class AddData extends Component {
-  state = {...DEFAULT_STATE};
-
-  onChange = (key, event) => this.setState({ [key]: event.target.value });
-
-  onSubmit = event => {
-    event.preventDefault();
-
-    this.props.doctorAdded({...this.state});
-    this.setState(DEFAULT_STATE);
-  };
-
-  render() {
-    return (
-      <div>
-        <form className="doctors" onSubmit={this.onSubmit}>
-          <p>Dodaj lekarza:</p>
-          <input
-            value={this.state.firstname}
-            onChange={this.onChange.bind(this, "firstname")}
-          />
-          <input
-            value={this.state.surname}
-            onChange={this.onChange.bind(this, "surname")}
-          />
-          <input
-            value={this.state.vetClinic}
-            onChange={this.onChange.bind(this, "vetClinic")}
-          />
-          <button>Zapisz</button>
-        
-        </form>
-      </div>
+const AddData = () => {
+  const [data, setData] = useState('');
+  const [firstname, setFirstname] = useState('');
+  const [surname, setSurname] = useState('');
+  const [vetClinic, setVetClinic] = useState('');
 
 
 
+  function onsubmit(e) {
+    e.preventDefault()
 
+    firebase.firestore()
+      .collection('doctors')
+      .add({
+        firstname,
+        surname,
+        vetClinic
+      })
 
+      .then(() => {
+        setFirstname('')
+        setSurname('')
+        setVetClinic('')
 
 
+      }
+      )
+  }
+  const { t } = useTranslation();
 
+  function handleClick(lang) {
+    i18next.changeLanguage(lang)
+    }
+  return (
+    <Modal trigger={<Button>{t('Dodaj lekarza.32')}</Button>}>
+      <Modal.Content>
+        <Modal.Description>
+          <Header>{t('Dodaj lekarza.32')}</Header>
 
+          <div>
+            <form className="doctors" onSubmit={onsubmit}><div>
+              <p>{t('Imię.2')}:</p>
+              <input value={firstname} onChange={e => setFirstname(e.currentTarget.value)}></input></div>
```

Comment:
same as previously - please move it to function

Reply:
done